### ISSUE TITLE
"sim" command (launch iOS Simulator) broken on Mavericks

### ISSUE THREAD

**Issue by michaelhogg** [AppGyver/steroids#1002]
This is a follow-up to #952.

### Before the bug (v4.1.4)

In [v4.1.4](https://github.com/AppGyver/steroids/releases/tag/v4.1.4), `package.json` [required `ios-sim-3.0.0-AppGyver-p1`](https://github.com/AppGyver/steroids/blob/v4.1.4/package.json#L63). That version of [`AppGyver/ios-sim`](https://github.com/AppGyver/ios-sim) (the AppGyver fork of [`phonegap/ios-sim`](https://github.com/phonegap/ios-sim)) was [released in November 2014](https://github.com/AppGyver/ios-sim/commit/a7e33ee).

### Introduction of the bug (v4.1.5)

One of the PRs included in [v4.1.5](https://github.com/AppGyver/steroids/releases/tag/v4.1.5) is #935, which updated `package.json` to require [`ios-sim-3.1.1-AppGyver-p1`](https://github.com/AppGyver/ios-sim/commit/9c44261).

Between version `3.0.0-AppGyver-p1` and `3.1.1-AppGyver-p1`, the latest version of the [3.x branch](https://github.com/phonegap/ios-sim/tree/3.x) of `phonegap/ios-sim` was [merged in](https://github.com/AppGyver/ios-sim/commit/1202df1).

Unfortunately, a bug had been introduced on the 3.x branch. (phonegap/ios-sim#145 was the cause, and I've raised phonegap/ios-sim#170 for the bug.)

`ios-sim-3.1.1-AppGyver-p1` now contains that bug, which causes ios-sim to crash on Mavericks if the `--devicetypeid` option is used.

That option is [used by `src/steroids/Simulator.coffee`](https://github.com/AppGyver/steroids/blob/v4.1.5/src/steroids/Simulator.coffee#L99) for launching the iOS Simulator when the `sim` command is run in the Steroids CLI.

So with v4.1.5 or later, running `steroids connect --debug` followed by `sim` causes this error:

```
Starting iPhone-6 Simulator

[DEBUG] Thu Jul 30 2015 13:36:19 GMT+0100 (BST) Killed iOS Simulator.
[DEBUG] Thu Jul 30 2015 13:36:19 GMT+0100 (BST) Spawning /Users/michaelhogg/.nvm/versions/node/v0.12.7/lib/node_modules/steroids/node_modules/ios-sim/bin/ios-sim
[DEBUG] Thu Jul 30 2015 13:36:19 GMT+0100 (BST) with params: launch,/Users/michaelhogg/.nvm/versions/node/v0.12.7/lib/node_modules/steroids/node_modules/steroids-ios-packages/simulators/Simulator-4.0.3.app,--devicetypeid,com.apple.CoreSimulator.SimDeviceType.iPhone-6,--verbose
2015-07-30 13:36:19.260 ios-sim[20989:507] -[__NSCFString containsString:]: unrecognized selector sent to instance 0x7fb18340d690

2015-07-30 13:36:19.261 ios-sim[20989:507] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFString containsString:]: unrecognized selector sent to instance 0x7fb18340d690'
*** First throw call stack:
(
    0   CoreFoundation                      0x00007fff9139925c __exceptionPreprocess + 172
    1   libobjc.A.dylib                     0x00007fff8d2ade75 objc_exception_throw + 43
    2   CoreFoundation                      0x00007fff9139c12d -[NSObject(NSObject) doesNotRecognizeSelector:] + 205
    3   CoreFoundation                      0x00007fff912f7272 ___forwarding___ + 1010
    4   CoreFoundation                      0x00007fff912f6df8 _CF_forwarding_prep_0 + 120
    5   ios-sim                             0x000000010fef6571 -[iPhoneSimulator runWithArgc:argv:] + 1245
    6   ios-sim                             0x000000010fef6e7c main + 101
    7   ios-sim                             0x000000010fef4140 start + 52
    8   ???                                 0x0000000000000006 0x0 + 6
)
libc++abi.dylib: terminating with uncaught exception of type NSException

[DEBUG] Thu Jul 30 2015 13:36:19 GMT+0100 (BST) Killing iOS Simulator ...
[DEBUG] Thu Jul 30 2015 13:36:19 GMT+0100 (BST) Killed iOS Simulator.
```

To summarise: Steroids v4.1.5 and later uses `ios-sim-3.1.1-AppGyver-p1`, which contains [this bug](https://github.com/phonegap/ios-sim/issues/170).

### Solution

The latest 3.x version of `phonegap/ios-sim` ([v3.2.0](https://github.com/phonegap/ios-sim/releases/tag/3.2.0)) still contains the bug, but [v4.1.1](https://github.com/phonegap/ios-sim/releases/tag/4.1.1) does not.

So, assuming you'd like to fix this bug on Mavericks, three possible solutions are:

 1. Fix the bug directly in `AppGyver/ios-sim`.
 2. Hope that a new 3.x version of `phonegap/ios-sim` will be released which fixes the bug, then merge it into `AppGyver/ios-sim`.
 3. Merge v4.1.1 of `phonegap/ios-sim` into `AppGyver/ios-sim`.

**Comment by harsaharsa** [comment]
Thanks for reporting this in such awesome detail; we'll see what can be done to fix the issue.
